tests/benchmarks/test_encoding.py

```diff
@@ -77,3 +81,26 @@ def test_join_encoded_custom(benchmark):
     benchmark(
         trace_encoder.join_encoded, [trace_encoder.encode_trace(trace_large), trace_encoder.encode_trace(trace_small)]
     )
+
+
+@pytest.mark.benchmark(group="encoding.dd_origin", min_time=0.005)
+def test_dd_origin_tagged_via_trace_tags_processor(benchmark):
```

Comment:
we should try best/worse case scenarios here

that is, maybe 1 scenario each for:

1. 1 trace with 1 span
2. 1 trace with 50 spans
3. 1 trace with 200 spans
4. 1 trace with 2000 spans

What do you think?